src/Disciplina/Educator/Mode.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Module contains the definition of Educator's WorkMode and its implementations.
+
+module Disciplina.Educator.Mode
+    (
+      -- * Constraints
+      EducatorWorkMode
+
+      -- * Implementations
+    , EducatorContext (..)
+    , EducatorRealMode
+    ) where
+
+import Universum
+
+import Control.Lens (makeLenses)
+
+import Disciplina.DB.Real.Types (NodeDB)
+import qualified Disciplina.Witness.Mode as Witness
+import Ether.Internal (HasLens (..))
+import System.Wlog (HasLoggerName (..))
+
+---------------------------------------------------------------------
+-- WorkMode class
+---------------------------------------------------------------------
+
+-- | Set of typeclasses which define capabilities of Educator node.
+type EducatorWorkMode m =
+    ( Witness.WitnessWorkMode m
```

Comment:
Nope. `EducatorWorkMode` should include only `BasicWorkMode` plus Educator-specific stuff, like SQL interface. This mode will be used for all Educator-specific logic. Also, we'll have a separate mode (like `CombinedWorkMode` or stuff) for pieces of logic which needs both.